presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1987,8 +1994,28 @@ public String getDescription()
 
         public void run(ExtendedHiveMetastore metastore)
         {
-            metastore.createTable(table, privileges);
-            done = true;
+            try {
+                metastore.createTable(table, privileges);
+                done = true;
+            }
+            catch (Throwable t) {
```

Comment:
Let's make this `PrestoException`. We definitely don't want to catch `Error`, and any other `RuntimeException` would be a weird error that shouldn't happen.